### PR TITLE
create a new lab: nifty

### PR DESCRIPTION
**Pull request by vipinsun**
nifty will be a low-code, metaprogrammable, standards sensitive and highly secure implementation to issue and trade NFTs, in a protocol agnostic way. 
nifty code will address market place utilities for issuance and trading of NFTs. 

Signed-off-by: Vipin Bharathan <vipinsun@gmail.com>

### CODE REVIEW

File: labs/nifty.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+For NFTs, the definition of fungibility is negated; that is, each token is unique and one cannot be exchanged for another. 
+
+A software system for the creation and trading for NFTs has to account for a way to tie the NFT to the unique object at issuance (Identity, authorization and a global registry); a marketplace for trading NFTs, which means establishing a price for the NFT as well as to transfer ownership. This means an implementation of these actions through interfaces supported by NFT token attributes and metadata.
+
+The following is the initial plan. 
+  Adopt a token standard (a combination of the TTF and ERC- inspired standards)  
+  Implement a marketplace (recode fabric samples )
+  Use Hyperledger Fabric as the underlying utility. See how ERC standards have been implemented in Fabric.
+  A complex systems view (Identity, distribution, decentralization, fairness, payment systems, cross-border, multi-protocol, interoperability of tokens, standards)- See under extensions.
```
Comment (tkuhrt) [inline labs/nifty.md:27]: Recommend adding `*` in front of these lines so that they read as bullets.

File: labs/nifty.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+Some ideas that will be explored
+    Can this infrastructure be flexible enough to define and issue different kinds of NFTs that conform to different standards? (Smart contract meta-programming)
+    Can metaprogramming in a multi-protocol approach work (this is considerably more difficult as it involves creating smart contracts in different protocols)- 
+    Investigate WASM approaches (investigate Solang HL Lab)
+    Interoperability of NFTs can NFT issued on one protocol move to another? After being paid for in another protocol
+    Oracles that underlie NFTs.
+    Submarining- Is this relevant for non-IPFS storage mechanisms? What about storj?
+    Other damping mechanisms- A governor or clamping mechanism with a circuit breaker for extreme events
+    Incorporate AI and XR in the stack
```
Comment (tkuhrt) [inline labs/nifty.md:39]: Recommend adding `*` in front of these lines so that they read as bullets.

File: labs/nifty.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+   Handle Current limitations of Web3 implementations (Moxie M)
+        "Design systems that can distribute trust without having to distribute infrastructure." . All data to be in distributed infrastructure (In Fabric this can be Private Data Collections)
+        "Reduce the burden of building software." - No code platform....A smart contract generator, for Go implementations first.
+        A generic issuer front end.
+        A wallet front end for transfers (not tied to a particular access point)
+        Platforms vs Protocols.
+        Avoid the worst of both worlds: "centralize to control, decentralize to reduce speed of change".
+        Wallets need interfaces to decentralized data without having to rely on a centralized intermediary.
```
Comment (tkuhrt) [inline labs/nifty.md:48]: Recommend adding `*` in front of these lines so that they read as bullets.

**Review by tkuhrt — APPROVED** [review]
I left a few cleanup comments, but other than that, this looks like a good lab.

File: labs/nifty.md (hunk 3 above)
Comment (vipinsun) [reply labs/nifty.md:48]: Will do so (add stars) inside our readme.md file...

Comment (vipinsun) [reply labs/nifty.md:48]: Any other pathway instead of recommitting and going through the PR process again @tkuhrt ?

Comment (tkuhrt) [reply labs/nifty.md:48]: Not that I am aware of.

**Comment by vipinsun** [comment]
@lehors @trbs please review. There is heavy interest across multiple SIGs and WGs about this lab.

**Review by lehors — APPROVED** [review]
I think some of the goals such as enabling exchange across networks/protocols are going to be very challenging. You will probably need to leverage interoperability technologies like Cactus or Weaver to achieve that without introducing any centralization. But LGTM.

File: labs/nifty.md (hunk 1 above)
Comment (IgorKrum) [reply labs/nifty.md:27]: Can this infrastructure protect against the scam?

**Pull request merged**